### PR TITLE
Extract a scope from all incoming credential types

### PR DESCRIPTION
**Pull request by alavers**
In the browser, scope is forced to 'appUser'. In node only, incoming
auth.jwt param will be rejected if its malformed or missing scope.

Take two of https://github.com/smooch/smooch-core-js/pull/53. The scope I'll be using later when adding managed account APIs. I've also cherry picked various  cleanup fixes from the closed PR.

### CODE REVIEW

File: tests/specs/utils/http.spec.js [hunk 1]
```diff
@@ -114,18 +114,22 @@ describe('HTTP', () => {
             });
         });
 
+        function randomIntBetween(min, max) {
+            return Math.floor(Math.random() * (max - min)) + min;
+        }
+
         describe('#handleStatus', () => {
-            for (let status = 200; status < 300; status++) {
+            [200, randomIntBetween(201, 299), 299].forEach((status) => {
```
Comment (lemieux) [inline tests/specs/utils/http.spec.js]: why the random instead of all of them?

Comment (alavers) [inline tests/specs/utils/http.spec.js]: The test output was bothering me. Iterating over every status code is unnecessary and actually introduced a few seconds slowdown.

Before:
```
 602 passing (3s)
```

After:
```
122 passing (423ms)
```

But boundary values + random compromise s a bit random. maybe I should just get rid of the random tests?

Comment (lemieux) [inline tests/specs/utils/http.spec.js]: /shrug I dunno, I don't like randomness in tests since you could *potentially* miss a value that would break the tests. As for test output, I personnally don't mind long test suite since it's the CI that should be running it mostly and locally you can `.only` the part you need.

Comment (alavers) [inline tests/specs/utils/http.spec.js]: I'll get rid of the random tests, but I'm not convinced of the necessity of testing all status codes between 200 and 599. Apart from special cases like 202 and 204, they don't represent distinct behaviors in the code.

When you get down to it, the test is ensuring our correct use of `>=` and `<` operators and their operands, and defending against regressions that change these bounds. To sufficiently validate this, you can simply need to check the boundaries, not every value in between.

Instead of the random values, I'll use a few common standard HTTP status codes instead. What do you think?

Comment (lemieux) [inline tests/specs/utils/http.spec.js]: works for me

Comment (mspensieri) [inline tests/specs/utils/http.spec.js]: Disclaimer: I'm just commenting to play devil's advocate, I don't really care if you do it this way.

> When you get down to it, the test is ensuring our correct use of `>=` and `<` operators and their operands, and defending against regressions that change these bounds.

This assumes that the underlying implementation is actually using those operators, or some form of range builder. A unit test shouldn't make assumptions about the underlying implementation. For example, the code could have been implemented as `if(statusCode === 200 || statusCode === 299)` and your test would pass even though `201` - `298` are not correctly handled

Comment (lemieux) [inline tests/specs/utils/http.spec.js]: which is basically the case because 202 and 204 are special cases for the body handling part.

Comment (alavers) [inline tests/specs/utils/http.spec.js]: To the devil I would say: as it stands if the code were to be implemented as `if(statusCode === 200 || statusCode === 299)` unit tests would break :)

</🚲 🏠 >
